source/core/Variants.cpp

```diff
@@ -55,14 +55,23 @@ VIREO_FUNCTION_SIGNATURET(VariantToData, VariantToDataParamBlock)
 {
     ErrorCluster *errPtr = _ParamPointer(ErrorClust);
     if (!errPtr || !errPtr->status) {
-        TypeRef variant = _Param(Variant);
+        TypeRef inputType = _ParamImmediate(InputData._paramType);
+        void* inputData = _ParamImmediate(InputData)._pData;
+
         TypeRef destType = _ParamImmediate(DestData._paramType);
+        void* destData = _ParamImmediate(DestData)._pData;
 
-        if (variant->IsA(destType)) {
-            void* destData = _ParamImmediate(DestData)._pData;
-            variant->CopyData(variant->Begin(kPARead), destData);
+        if (inputType->Name().Compare(&TypeCommon::TypeVariant)) {
```

Comment:
This is definitely an improvement but it doesn't handle compatible type conversion, which Variant to Data also does.
For example, if the input type is an Int32 and the output type is another numeric, it should do the conversion.
Also, if the output type is a variant and the input type isn't, that should work too, converting the input data to a variant (bizarrely, this makes the node act like Data to Variant, its own inverse).  This happens in the source model when the 'type' input is unwired, as it defaults to variant.

Reply:
We have a work item tracking the first case you mention. For the second one, I brought it up during standup today, and it sounded like something we did not want to carry forward on Web VIs -- the team leaned towards reporting an error if the DataType terminal is unwired. I will track the outcome of that discussion as a separate work item.